contracts/auctions/src/contract.rs

```diff
@@ -0,0 +1,430 @@
+            &auction.item_info.item_id,
+            &1,
+        );
+
+        auction.status = AuctionStatus::Ended;
+
+        update_auction(&env, auction_id, auction.clone())?;
+        save_auction_by_seller(&env, &auction.seller, &auction)?;
+        save_auction_by_id(&env, auction_id, &auction)?;
+
+        Ok(())
+    }
+
+    #[allow(dead_code)]
+    pub fn buy_now(env: Env, auction_id: u64, buyer: Address) -> Result<(), ContractError> {
+        buyer.require_auth();
+
+        let mut auction = get_auction_by_id(&env, auction_id)?;
+
+        if env.ledger().timestamp() > auction.end_time || auction.status != AuctionStatus::Active {
+            log!(env, "Auction: Buy now: Auction not active");
+            return Err(ContractError::AuctionNotActive);
+        }
+
+        if auction.item_info.buy_now_price.is_none() {
+            log!(
+                env,
+                "Auction: Buy now: trying to buy an item that does not allow `buy now`"
+            );
+            return Err(ContractError::NoBuyNowOption);
+        }
+
+        let old_highest_bid = get_highest_bid(&env, auction_id)?;
+
+        // refund any previous highest bid
+        let token = token::Client::new(&env, &auction.currency);
+
+        token.transfer(
+            &env.current_contract_address(),
+            &old_highest_bid.bidder,
+            &(old_highest_bid.bid as i128),
+        );
+
+        // pay for the item
+        token.transfer(
+            &buyer,
+            &auction.seller,
+            &(auction
+                .item_info
+                .buy_now_price
+                .expect("Buy now price has not been set") as i128),
+        );
+
+        let collection_client = collection::Client::new(&env, &auction.item_info.collection_addr);
+
+        collection_client.safe_transfer_from(
+            &auction.seller,
+            &buyer,
+            &auction.item_info.item_id,
+            &ONE,
+        );
+
+        auction.status = AuctionStatus::Ended;
+        auction.highest_bidder = buyer;
+        auction.highest_bid = Some(
+            auction
+                .item_info
+                .buy_now_price
+                .expect("Buy now price has not been set"),
+        );
+
+        update_auction(&env, auction_id, auction.clone())?;
+        save_auction_by_id(&env, auction_id, &auction)?;
+        save_auction_by_seller(&env, &auction.seller, &auction)?;
+
+        Ok(())
+    }
+
+    #[allow(dead_code)]
+    pub fn pause(env: Env, auction_id: u64) -> Result<(), ContractError> {
+        let mut auction = get_auction_by_id(&env, auction_id)?;
+        auction.seller.require_auth();
+
+        if auction.status != AuctionStatus::Active {
+            log!(env, "Auction: Pause: Cannot pause inactive/ended auction.");
+            return Err(ContractError::AuctionNotActive);
+        }
+
+        if env.ledger().timestamp() > auction.end_time {
+            log!(env, "Auction: Buy now: Auction expired");
```

Comment:
```suggestion
            log!(env, "Auction: Pause: Auction expired");
```

Reply:
fixed a few others in that lines of thought